projects/packages/my-jetpack/_inc/components/product-cards-section/boost-card/boost-speed-score.tsx

```diff
@@ -0,0 +1,139 @@
+import { getScoreLetter, requestSpeedScores } from '@automattic/jetpack-boost-score-api';
+// We'll need the IconTooltip in a follow-up PR. TODO: remove me
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { IconTooltip, Spinner, BoostScoreBar } from '@automattic/jetpack-components';
+import { __, sprintf } from '@wordpress/i18n';
+import React, { useCallback, useEffect, useState } from 'react';
+import useAnalytics from '../../../hooks/use-analytics';
+import './style.scss';
+
+const BoostSpeedScore = () => {
+	const { recordEvent } = useAnalytics();
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ speedLetterGrade, setSpeedLetterGrade ] = useState( 'C' );
+	const [ daysSinceTested, setDaysSinceTested ] = useState( 1 );
+	const [ averageSpeedScore, setAverageSpeedScore ] = useState( 0 );
+	const [ isSpeedScoreError, setIsSpeedScoreError ] = useState( false );
+
+	const { apiRoot, apiNonce, connectionStatus } = window.JP_CONNECTION_INITIAL_STATE;
+	const { siteSuffix: siteUrl = '', latestBoostSpeedScores } = window?.myJetpackInitialState ?? {};
+	const isSiteRegistered = connectionStatus?.isRegistered;
+
+	const setScoresFromCache = cachedSpeedScores => {
+		setAverageSpeedScore(
+			Math.round( ( cachedSpeedScores.scores.mobile + cachedSpeedScores.scores.desktop ) / 2 )
+		);
+		setSpeedLetterGrade(
+			getScoreLetter( cachedSpeedScores.scores.mobile, cachedSpeedScores.scores.desktop )
+		);
+		setDaysSinceTested( calculateDaysSince( cachedSpeedScores.timestamp * 1000 ) );
+	};
+
+	const getSpeedScores = async () => {
+		if ( ! isSiteRegistered ) {
+			setIsSpeedScoreError( true );
+			return;
+		}
+
+		setIsLoading( true );
+		try {
+			const scores = await requestSpeedScores( true, apiRoot, siteUrl, apiNonce );
+			const scoreLetter = getScoreLetter( scores.current.mobile, scores.current.desktop );
+			setSpeedLetterGrade( scoreLetter );
+			setAverageSpeedScore( Math.round( ( scores.current.mobile + scores.current.desktop ) / 2 ) );
```

Comment:
Might be overkill, but I see that Math.round function with the scores used above too. Might be better to be separate as a function named something like (getAverageSpeedScore)? That's just a thought, not a strong opinion on my part

Reply:
Yeah, this would be nice. I like it, thanks, will do! 🙂